beater/validator.go

```diff
@@ -30,14 +30,12 @@ import (
 type validator struct{}
 
 func (v *validator) Validate(cfg *agentconfig.C) error {
-	c, err := config.New(cfg)
+	// TODO: Should we check something?
+	// Benchmark is being checked inside config.New
+	_, err := config.New(cfg)
```

Comment:
I feel like the validation inside config should be here

```go
if c.Benchmark != "" {
if !isSupportedBenchmark(c.Benchmark) {
	return c, ErrBenchmarkNotSupported
	}
}
```

Reply:
I've considered moving the validation to the validator but that would require us to move logic from the `config` package to the `beater` package.
Which requires recoding parts of the `cloudbeat` like `newCloudbeat` signatures and etc...
Reconsidering the existing solution I did not see any value in the `validator`, because we use the same validation for non-fleet managed configs and reconfigurations.
This is why I've consolidated the validity checks in the config constructor and removed the validator.